test/e2e/common_suite_test.go

```diff
@@ -237,6 +263,40 @@ func newTestCase(t *testing.T, testName string, assert CloudAssert, assessMessag
 	return testCase
 }
 
+func comparePodLogString(ctx context.Context, t *testing.T, client klient.Client, customPod v1.Pod, expectedPodlogString string) (bool, string, error) {
+	podLogString := ""
+	var podlist v1.PodList
+	clientset, err := kubernetes.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return false, "", err
+	}
+	if err := client.Resources(customPod.Namespace).List(ctx, &podlist); err != nil {
+		return false, "", err
+	}
+	for _, pod := range podlist.Items {
+		if pod.ObjectMeta.Name == customPod.Name {
+			func() {
+				req := clientset.CoreV1().Pods(customPod.Namespace).GetLogs(pod.ObjectMeta.Name, &v1.PodLogOptions{})
+				podLogs, err := req.Stream(ctx)
+				if err != nil {
+					return
+				}
+				defer podLogs.Close()
+				buf := new(bytes.Buffer)
+				_, err = io.Copy(buf, podLogs)
+				if err != nil {
+					return
+				}
+				podLogString = strings.TrimSpace(buf.String())
+			}()
+		}
+	}
+	if podLogString != expectedPodlogString {
+		return false, podLogString, nil
```

Comment:
Instead of returning `false` and a `nil` you can return a  new error (e.g., from `errors.New()`) here.

This also lets you simplify the types returned by  `comparePodLogString` to just `(string, error)`.